### PR TITLE
Map classes relating to candles

### PR DESCRIPTION
**Pull request by haykam821**
This pull request maps new additions relating to candles and candle cakes in 20w45a.

### CODE REVIEW

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 1]
```diff
@@ -559,6 +560,9 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 		METHOD method_26234 isFullCube (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Z
 			ARG 1 world
 			ARG 2 pos
+		METHOD method_27851 testIfInTag (Lnet/minecraft/class_3494;Ljava/util/function/Predicate;)Z
```
Comment (YanisBft) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		METHOD method_27851 isIn (Lnet/minecraft/class_3494;Ljava/util/function/Predicate;)Z
```
`isIn` should suffice

File: mappings/net/minecraft/block/AbstractCandleBlock.mapping [hunk 2]
```diff
@@ -0,0 +1,19 @@
+CLASS net/minecraft/class_5540 net/minecraft/block/AbstractCandleBlock
+	FIELD field_27083 LIT Lnet/minecraft/class_2746;
+	METHOD method_31610 spawnCandleParticles (Lnet/minecraft/class_1937;Lnet/minecraft/class_243;Ljava/util/Random;)V
+		ARG 0 world
+		ARG 1 vec3d
```
Comment (YanisBft) [inline mappings/net/minecraft/block/AbstractCandleBlock.mapping:5]:
```suggestion
		ARG 1 offset
```

File: mappings/net/minecraft/block/CandleCakeBlock.mapping [hunk 3]
```diff
@@ -0,0 +1,18 @@
+CLASS net/minecraft/class_5545 net/minecraft/block/CandleCakeBlock
+	FIELD field_27183 LIT_2 Lnet/minecraft/class_2746;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/CandleCakeBlock.mapping]:
```suggestion
	FIELD field_27183 LIT Lnet/minecraft/class_2746;
```

File: mappings/net/minecraft/block/CandleBlock.mapping [hunk 4]
```diff
@@ -0,0 +1,16 @@
+CLASS net/minecraft/class_5544 net/minecraft/block/CandleBlock
+	FIELD field_27174 CANDLES Lnet/minecraft/class_2758;
+	FIELD field_27175 LIT_2 Lnet/minecraft/class_2746;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/CandleBlock.mapping]:
```suggestion
	FIELD field_27175 LIT Lnet/minecraft/class_2746;
```

File: mappings/net/minecraft/block/AbstractCandleBlock.mapping (hunk 2 above)
Comment (haykam821) [reply mappings/net/minecraft/block/AbstractCandleBlock.mapping:5]: That is the actual position of the particle, and is calculated by `offset.add(pos.getX(), pos.getY(), pos.getZ())`, where `offset` is an offset `Vec3d` provided by `getParticleOffsets `/`method_31613` and `pos` is the position of a candle's display tick.

**Review by OroArmor — APPROVED** [review]

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 5]
```diff
@@ -296,6 +296,7 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 			ARG 1 predicate
 		METHOD method_26250 air ()Lnet/minecraft/class_4970$class_2251;
 		METHOD method_29292 requiresTool ()Lnet/minecraft/class_4970$class_2251;
+		METHOD method_31710 materialColor (Lnet/minecraft/class_3620;)Lnet/minecraft/class_4970$class_2251;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		METHOD method_31710 mapColor (Lnet/minecraft/class_3620;)Lnet/minecraft/class_4970$class_2251;
```

File: mappings/net/minecraft/block/CandleBlock.mapping (hunk 4 above)
Comment (haykam821) [reply mappings/net/minecraft/block/CandleBlock.mapping]: Fixed in 48c14bf52cc9898db151ecdd2b02c0851839a51a

File: mappings/net/minecraft/block/CandleCakeBlock.mapping (hunk 3 above)
Comment (haykam821) [reply mappings/net/minecraft/block/CandleCakeBlock.mapping]: Fixed in 48c14bf52cc9898db151ecdd2b02c0851839a51a

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 1 above)
Comment (haykam821) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: Fixed in 3ce10eccb2f0ed761fbf6162a71e12797f95f234

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 5 above)
Comment (haykam821) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: Fixed in c370022e4da786882264e042b878a59d2f569d2a

**Review by YanisBft — APPROVED** [review]

**Review by jaskarth — APPROVED** [review]

**Pull request merged**